File: Learning Codes/Linked List/insert_at_head.py
class Node:
    def linked_list(self, val):
        self.data = val
        self.next = None

def insert_at_head(head, val):
    n = Node()
    n.linked_list(val)
    n.next = head
    head = n
    return head

def display(head):
    tmp = head
    while tmp is not None:
        print(tmp.data, "-->", end=" ")
        tmp = tmp.next
    print("NULL")

head = None
display(head)
head = insert_at_head(head, 1)
head = insert_at_head(head, 10)
head = insert_at_head(head, 100)
head = insert_at_head(head, 1000)
display(head)

File: Learning Codes/Linked List/Mine/C++/insert_at_head.cpp
#include <bits/stdc++.h>
using namespace std;

class node
{
public:
    int data;
    node *next;

    node(int val)
    {
        data = val;
        next = NULL;
    }
};

void insert_at_head(node *&head, int val)
{
    node *n = new node(val);
    n->next = head;
    head = n;
}
void display(node *head)
{
    node *tmp = head;
    while (tmp != NULL)
    {
        cout << tmp->data << " --> ";
        tmp = tmp->next;
    }
    cout << "NULL" << endl;
}

int main()
{
    node *head = NULL;
    display(head);
    insert_at_head(head, 2);
    insert_at_head(head, 20);
    insert_at_head(head, 200);
    display(head);
    return 0;
}
